internal/util/validation/services.go

```diff
@@ -35,6 +37,12 @@ var (
 func ServicesHaveValidTemplates(ctx context.Context, cl client.Client, services []kcmv1.Service, ns string) error {
 	var errs error
 	for _, svc := range services {
+		if svc.Namespace != "" {
+			for _, msg := range validation.IsDNS1123Label(svc.Namespace) {
```

Comment:
There is also [`ValidateNamespaceName`](https://github.com/kubernetes/apimachinery/blob/84d740c9e27f3ccc94c8bc4d13f1b17f60f7080b/pkg/api/validation/generic.go#L66-L69) func which is an indirect alias to this one but may be better to use as its name is more indicative that it should be used to verify namespace.